### ISSUE TITLE
Using my own dataset

### ISSUE THREAD

**Issue by Revol123** [filaPro/oneformer3d#64]
Thank you for your great work on this project. I am currently working on my own indoor dataset in USDZ format, and I'd like to test it with your pre-trained model. However, I'm not sure what the next step should be.

Should I convert my dataset into a format like ScanNet or S3DIS? If so, which one is better?
Alternatively, would it be feasible to modify [test.py](https://github.com/oneformer3d/oneformer3d/blob/main/tools/test.py) or even write a new tester to test my dataset? I'd appreciate any recommendations you might have.

Thank you for your assistance.

**Comment by oneformer3d-contributor** [comment]
Yes, converting to S3DIS format should be fine. We already got comments, that it works fine.
For ScanNet we use unsupervised superpoint clustering, and probably you don't want to figure it out for your dataset.

**Comment by Revol123** [comment]
Thanks for your response. I have another question now. For example, in the S3DIS dataset structure:
Area_1
└── conferenceRoom_2
    ├── Annotations
    │   ├── Icon
    │   ├── beam_1.txt
    │   ├── board_1.txt
    │   └── the other txt files
    ├── Icon
    └── conferenceRoom_2.txt

I've obtained all the points with colors, like in conferenceRoom_2.txt above. But it seems there are no labels for each point in my dataset, so I don't have files in the Annotations folder. Is there another way I can test my dataset without using annotation files in your code? Or if you have any other recommendations, I would greatly appreciate it!

**Comment by oneformer3d-contributor** [comment]
Actually data loading pipeline in `test.py` should not require gt annotations. Just comment the evaluation call at some point and it should be fine.

**Comment by D3xter1922** [comment]
I was trying to train oneformer with my own dataset. I have converted it into the S3DIS format. Training goes fine on a 24GB card however it runs out of VRAM on validation step. Batch size for both is 1. Is there a config parameter I am missing here?

**Comment by oneformer3d-contributor** [comment]
> I was trying to train oneformer with my own dataset. I have converted it into the S3DIS format. Training goes fine on a 24GB card however it runs out of VRAM on validation step. Batch size for both is 1. Is there a config parameter I am missing here?

Yes, the solution is in our readme.

**Comment by oneformer3d-contributor** [comment]
It is strange to run out of GPU memory on validation. Because if it is CPU memory we recommend to increase `inst_score_thr` or decrease `topk_insts`. For GPU memory you can try to decrease `num_points`.

**Comment by D3xter1922** [comment]
Thank you very much for the help. It worked and I added a `PointSample` in my test_pipeline, however, during evaluation, it is not considering those sampled points during evaluation, and I get the error that `gt_labels` and `seg_labels` are of different lengths. Is there a configuration option for that?(to make sure evaluation also considers the same sampled points?) Thanks!

**Comment by LiNY2-01** [comment]
> For ScanNet we use unsupervised superpoint clustering, and probably you don't want to figure it out for your dataset.

Hi! Thank you for your great work on this project. I am currently working with my own indoor point cloud data, and I would like to infer it using your pre-trained model on the ScanNet200 dataset. I am confused about how to apply superpoint clustering, as I only have points and their colors, and no mesh in my point cloud. Should I perform surface reconstruction, or should I just segment the points directly? If you have any recommendations, I would greatly appreciate it!
